### PR TITLE
HWP-375 Added maintenance page

### PR DESCRIPTION
**Pull request by ZachBourque**
# Description

This PR adds a maintenance page which is redirected to if the API is down.

This PR includes the following proposed change(s):

- Inclusion of the helperFunctions folder and isAPIHealthy.js
- Addition of APIDownPage.jsx
- Modified Routes.jsx to conditionally 

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/client/src/views/APIDownPage.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Box, Grid, Typography, Stack, Tooltip } from "@mui/material";
+import React from "react";
+
+function APIDownPage() {
+  //TODO: Re-call the API ever X seconds to automatically redirect the user if the API is up
```
Comment (BradyMitch) [inline app/client/src/views/APIDownPage.jsx:5]: Yes this will be a very nice touch!

File: app/client/src/views/APIDownPage.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <Stack
+      alignItems="center"
+      justifyContent="space-around"
+      style={{ height: "85vh" }}
+      spacing={2}
+    >
+      <Stack alignItems="center" spacing={2}>
+        <Typography variant="h4">Sorry, HWP is down for maintenance</Typography>
```
Comment (BradyMitch) [inline app/client/src/views/APIDownPage.jsx:15]:
```suggestion
        <Typography variant="h4">Sorry, HWP is down for maintenance.</Typography>
```
```suggestion
        <Typography variant="h4">Sorry, HWP is down for maintenance</Typography>
```

File: app/client/src/views/APIDownPage.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <Typography variant="h6">Please check back soon</Typography>
```
Comment (BradyMitch) [inline app/client/src/views/APIDownPage.jsx:16]:
```suggestion
        <Typography variant="h6">Please check back soon.</Typography>
```
```suggestion
        <Typography variant="h6">Please check back soon</Typography>
```

**Review by BradyMitch — COMMENTED** [review]
Excellent addition! I did find that if the API is down when you are navigating with the nav menu it doesn't redirect to the maintenance page. Is this something that could be fixed?

**Comment by ZachBourque** [comment]
> Excellent addition! I did find that if the API is down when you are navigating with the nav menu it doesn't redirect to the maintenance page. Is this something that could be fixed?

That's a great point. I'll create a ticket for that change.

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**